### PR TITLE
[Merged by Bors] - Document exotic patterns for `Command`s and `Event`s

### PR DESCRIPTION
**Pull request by JoJoJet**
# Objective

Improve documentation, information users of the limitations in bevy's idiomatic patterns, and suggesting alternatives for when those limitations are encountered.

## Solution

* Add documentation to `Commands` informing the user of the option of writing one-shot commands with closures.
* Add documentation to `EventWriter` regarding the limitations of event types, and suggesting alternatives using commands.

### CODE REVIEW

**Review by DJMcNab — COMMENTED** [review]
Implementation is fine. I'm not sure I want us to encourage this though.

If users want to do this themselves, they should. I guess you could add this as an example to the docs of `impl Command for F where F: FnMut(&mut World)` - that way we can still show people it's possible, but make it clear that it's not idiomatic.

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
I agree with @DJMcNab above; can you adapt this PR to show the untyped event pattern as a doc example on either `Events` or `EventWriter`?

**Review by MinerSebas — CHANGES_REQUESTED** [review]
Please update the PR Title and Description to reflect that this is now a doc change.

File: crates/bevy_ecs/src/event.rs [hunk 1]
```diff
@@ -248,7 +248,48 @@ impl<'w, 's, E: Event> EventReader<'w, 's, E> {
...
+///
+/// pub struct MyEvent; // Custom event type.
+/// fn my_system(mut writer: EventWriter<MyEvent>) {
+///     writer.send(MyEvent);
+/// }
+///
+/// # bevy_ecs::system::assert_is_system(my_system);
+/// ```
+///
+/// # Limitations
```
Comment (alice-i-cecile) [inline crates/bevy_ecs/src/event.rs:267]:
```suggestion
/// # Limitations
///
```

**Review by alice-i-cecile — APPROVED** [review]
Changes LGTM now :)

File: crates/bevy_ecs/src/event.rs [hunk 2]
```diff
@@ -248,7 +248,48 @@ impl<'w, 's, E: Event> EventReader<'w, 's, E> {
     }
 }
 
-/// Sends events of type `T`.
+/// Sends events of type `T` to [`EventReader`]s of type `T`.
```
Comment (IceSentry) [inline crates/bevy_ecs/src/event.rs:251]: I know this makes sense from a high level perspective, but technically the events are sent to `Events<T>` and `EventReader` is just a nicer abstraction around that.

Personally I would simply not mention it at all

```suggestion
/// Sends events of type `T`.
```

**Review by IceSentry — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
@JoJoJet I'm ready to merge this, but the PR title should really be updated to ensure that our commit history makes sense.

**Comment by JoJoJet** [comment]
I believe that addresses everything.

**Review by alice-i-cecile — COMMENTED** [review]
bors r+

**Comment by alice-i-cecile** [comment]
bors retry

**Pull request closed without merging**